src/control/lib/txtfmt/table_test.go

```diff
@@ -107,53 +108,102 @@ func TestTableFormatter_Format(t *testing.T) {
-			table:          []TableRow{{"One": "1", "Two": "2", "Three": "3"}, {"One": "4", "Two": "5", "Three": "6"}},
-			expectedResult: "One\tTwo\tThree\t\n---\t---\t-----\t\n1\t2\t3\t\n4\t5\t6\t\n",
+			titles: []string{"One", "Two", "Three"},
+			table:  []TableRow{{"One": "1", "Two": "2", "Three": "3"}, {"One": "4", "Two": "5", "Three": "6"}},
+			expectedResult: `
+One Two Three 
+--- --- ----- 
+1   2   3     
+4   5   6     
+`,
 		},
 		"empty table": {
-			titles:         []string{"One"},
-			table:          []TableRow{},
-			expectedResult: "One\t\n---\t\n",
+			titles: []string{"One"},
+			table:  []TableRow{},
+			expectedResult: `
+One 
+--- 
+`,
 		},
 		"no matches for titles": {
-			titles:         []string{"One", "Two"},
-			table:          []TableRow{{}},
-			expectedResult: "One\tTwo\t\n---\t---\t\nNone\tNone\t\n",
+			titles: []string{"One", "Two"},
+			table:  []TableRow{{}},
+			expectedResult: `
+One  Two  
+---  ---  
+None None 
+`,
 		},
 		"matches for some titles": {
-			titles:         []string{"One", "Two"},
-			table:          []TableRow{{"One": "1"}, {"Two": "2"}},
-			expectedResult: "One\tTwo\t\n---\t---\t\n1\tNone\t\nNone\t2\t\n",
+			titles: []string{"One", "Two"},
+			table:  []TableRow{{"One": "1"}, {"Two": "2"}},
+			expectedResult: `
+One  Two  
+---  ---  
+1    None 
+None 2    
+`,
 		},
 		"extra keys ignored": {
-			titles:         []string{"One", "Three"},
-			table:          []TableRow{{"One": "1", "Two": "2", "Three": "3", "Four": "4"}},
-			expectedResult: "One\tThree\t\n---\t-----\t\n1\t3\t\n",
+			titles: []string{"One", "Three"},
+			table:  []TableRow{{"One": "1", "Two": "2", "Three": "3", "Four": "4"}},
+			expectedResult: `
+One Three 
+--- ----- 
+1   3     
+`,
 		},
 		"tabbing for long string": {
-			titles:         []string{"One", "Two"},
-			table:          []TableRow{{"One": "1", "Two": "2"}, {"One": "too darn long"}},
-			expectedResult: "One\t\tTwo\t\n---\t\t---\t\n1\t\t2\t\ntoo darn long\tNone\t\n",
+			titles: []string{"One", "Two"},
+			table:  []TableRow{{"One": "1", "Two": "2"}, {"One": "too darn long"}},
+			expectedResult: `
+One           Two  
+---           ---  
+1             2    
+too darn long None 
+`,
+		},
+		"8-char field should be padded": {
+			titles: []string{"Hosts", "SCM Total", "NVMe Total"},
+			table: []TableRow{{
+				"Hosts":      "wolf-118",
+				"SCM Total":  "5.79TB (2 namespaces)",
+				"NVMe Total": "1.46TB (2 controllers)",
+			}},
+			expectedResult: `
+Hosts    SCM Total             NVMe Total             
+-----    ---------             ----------             
+wolf-118 5.79TB (2 namespaces) 1.46TB (2 controllers) 
+`,
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
 			f := NewTableFormatter(tt.titles...)
 
 			result := f.Format(tt.table)
 
-			if result != tt.expectedResult {
-				t.Fatalf("want: '%s', got: '%s'", tt.expectedResult, result)
+			if diff := cmp.Diff(strings.TrimLeft(tt.expectedResult, "\n"), result); diff != "" {
```

Comment:
Might be some use in having a shared utility method to do this trimming for us in the same way in all cases, since we repeat this a few times in these formatting tests.

There is a part of me that also wants a common shared helper for doing the cmp.Diff comparison since the output is basically identical every time. Needn't be addressed here but maybe something to think about.

Reply:
Honestly, redundancy like this doesn't bother me too much in test code. It's an optimization for clarity for the reader who has to figure out why the test is failing. Adding helpers to save a few lines of code means that the reader has to go study the helper to figure out what's going on over there before coming back here.

Reply:
That's fair.